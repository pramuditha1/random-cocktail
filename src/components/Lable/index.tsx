import { Typography, TypographyProps } from "@mui/material";
import React from "react";

// Exporting the LablePropTypes interface
export interface LablePropTypes {
  text: string;
  variant: TypographyProps["variant"];
  component?: TypographyProps["component"];
  color?: TypographyProps["color"];
}

const Lable: React.FC<LablePropTypes> = ({
  text,
  variant = "body1",
  component = "div",
  color = "textPrimary",
}) => {
  return (
    <Typography
      gutterBottom
      variant={variant}
      component={component}
      color={color}
    >
      {text}
    </Typography>
  );
};

export default Lable;
