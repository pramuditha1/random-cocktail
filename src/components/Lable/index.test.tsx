import React from "react";
import { render } from "@testing-library/react";
import Lable, { LablePropTypes } from "./index";

test("Lable component matches snapshot", () => {
  const mockProps: LablePropTypes = {
    variant: "h1",
    text: "test",
  };

  const { asFragment } = render(<Lable {...mockProps} />);
  expect(asFragment()).toMatchSnapshot();
});
