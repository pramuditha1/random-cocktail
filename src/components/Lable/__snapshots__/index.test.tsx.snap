// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lable component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-79core-MuiTypography-root"
  >
    test
  </div>
</DocumentFragment>
`;
